File: code/collections/map.js
// Map (higher-order) in JavaScript
var numbers = [20, 19, 7, 12];
var tripled = numbers.map(function(x) { return x * 3; });
// Or with arrow function (ES6)
var doubled = numbers.map(x =&gt; x * 2);
console.log(tripled);
console.log(doubled);

File: code/collections/map.java
// Map (higher-order) in Java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Example {
    public static void main(String[] args) {
        List&lt;Integer&gt; numbers = Arrays.asList(20, 19, 7, 12);
        List&lt;Integer&gt; tripled = numbers.stream()
            .map(x -&gt; x * 3)
            .collect(Collectors.toList());
        List&lt;Integer&gt; doubled = numbers.stream()
            .map(x -&gt; x * 2)
            .collect(Collectors.toList());
        System.out.println(tripled);
        System.out.println(doubled);
    }
}
